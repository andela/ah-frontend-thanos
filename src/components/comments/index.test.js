import React from 'react';
import { shallow } from 'enzyme';
import { CommentForm, CommentsPage } from '.';

describe('<Article />', () => {
  const comments = {
    getCommentData: {
      results: [
        {
          id: 1,
          comment_body: 'This is a comment',
          comment_author:
            { id: 7, username: 'Sulaiman', email: 'sulaiman@andela.com' },
          created_at: '2018-12-10T05:23:51.342553Z',
          updated_at: '2018-12-10T05:23:51.342616Z',
        },
      ],
    },
  };

  test('should render correctly', () => {
    const wrapper = shallow(
      <CommentsPage comments={comments} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('should render correctly', () => {
    const wrapper = <CommentForm onSubmitHandler={jest.fn()} onChangeHandler={jest.fn()} />;
    const CommentWrapper = shallow(wrapper);
    expect(CommentWrapper.find('.col-sm-8').length).toBe(1);
    expect(CommentWrapper).toMatchSnapshot();
  });
});
