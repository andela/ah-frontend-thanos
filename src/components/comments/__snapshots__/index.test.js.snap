// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> should render correctly 1`] = `
<div
  className="col-sm-8 offset-2"
>
  <br />
  <div>
    <div
      key="1"
    >
      <div
        className="media border p-3 bg-light"
      >
        <div
          className="media-body"
        >
          <h6>
            Sulaiman
          </h6>
          <p>
            This is a comment
          </p>
          <small>
            <i>
              Posted on  â€‚
            </i>
            2018-12-10T05:23:51.342553Z
          </small>
        </div>
      </div>
      <br />
    </div>
  </div>
</div>
`;

exports[`<Article /> should render correctly 2`] = `
<div
  className="col-sm-8 offset-2"
>
  <h3>
     Comments 
  </h3>
  <br />
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="form-group"
    >
      <p>
        Write your comment :
      </p>
      <textarea
        className="form-control"
        id="comment"
        name="comment_body"
        onChange={[MockFunction]}
        required={true}
        rows="5"
      />
    </div>
    <button
      className="btn btn-dark offset-10"
      type="submit"
    >
      Submit
    </button>
  </form>
  <br />
  <br />
</div>
`;
